server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/PartitionCache.java

```diff
@@ -0,0 +1,710 @@
+      boolean result = true;
+      for (String devicePath : devicePaths) {
+        String storageGroup = getStorageGroup(devicePath);
+        if (null == storageGroup) {
+          logger.debug(
+              "[{} Cache] miss when search device {}", STORAGE_GROUP_CACHE_NAME, devicePath);
+          result = false;
+          if (failFast) {
+            break;
+          } else {
+            failedDevices.add(devicePath);
+          }
+        } else {
+          deviceToStorageGroupMap.put(devicePath, storageGroup);
+        }
+      }
+      if (!result) {
+        deviceToStorageGroupMap.clear();
+      }
+      logger.debug("[{} Cache] hit when search device {}", STORAGE_GROUP_CACHE_NAME, devicePaths);
+      CacheMetricsRecorder.record(result, STORAGE_GROUP_CACHE_NAME);
+      return result;
+    } finally {
+      storageGroupCacheLock.readLock().unlock();
+    }
+  }
+
+  /** get storage group to device map with failed devices */
+  public boolean getStorageGroupToDeviceMap(
+      List<String> devicePaths,
+      Map<String, List<String>> storageGroupToDeviceMap,
+      List<String> failedDevices,
+      boolean failFast) {
+    storageGroupCacheLock.readLock().lock();
+    try {
+      boolean result = true;
+      for (String devicePath : devicePaths) {
+        String storageGroup = getStorageGroup(devicePath);
+        if (null == storageGroup) {
+          logger.debug(
+              "[{} Cache] miss when search device {}", STORAGE_GROUP_CACHE_NAME, devicePath);
+          result = false;
+          if (failFast) {
+            break;
+          } else {
+            failedDevices.add(devicePath);
+          }
+        } else {
+          List<String> devices =
+              storageGroupToDeviceMap.computeIfAbsent(storageGroup, k -> new ArrayList<>());
+          devices.add(devicePath);
+        }
+      }
+      if (!result) {
+        storageGroupToDeviceMap.clear();
+      }
+      logger.debug("[{} Cache] hit when search device {}", STORAGE_GROUP_CACHE_NAME, devicePaths);
+      CacheMetricsRecorder.record(result, STORAGE_GROUP_CACHE_NAME);
+      return result;
+    } finally {
+      storageGroupCacheLock.readLock().unlock();
+    }
+  }
+
+  /** get storage group to device map in three trys */
+  public Map<String, List<String>> getStorageGroupToDevice(
+      List<String> devicePaths, boolean isAutoCreate) {
+    Map<String, List<String>> storageGroupToDeviceMap = new HashMap<>();
+    // miss when devicePath contains *
+    for (String devicePath : devicePaths) {
+      if (devicePath.contains("*")) {
+        return storageGroupToDeviceMap;
+      }
+    }
+    // first try to hit storage group in fast-fail way
+    boolean firstTryResult =
+        getStorageGroupToDeviceMap(devicePaths, storageGroupToDeviceMap, new ArrayList<>(), true);
+    if (!firstTryResult) {
+      try {
+        // when local cache not have, then try to fetch all storage group from config node
+        getAllStorageGroupFromConfigNode();
```

Comment:
Suggest to rename this method to some name such as `fetchStorageGroupAndUpdateCache()`

Reply:
Get it.